src/Avalonia.Base/Media/TextFormatting/TextCharacters.cs

```diff
@@ -115,16 +115,11 @@ private static UnshapedTextRun CreateShapeableRun(ReadOnlyMemory<char> text,
 
             var codepoint = Codepoint.ReplacementCodepoint;
 
-            var codepointEnumerator = new CodepointEnumerator(text.Slice(count).Span);
+            var codepointEnumerator = new GraphemeEnumerator(text.Slice(count).Span);
 
-            while (codepointEnumerator.MoveNext(out var cp))
+            while (codepointEnumerator.MoveNext(out var grapheme))
```

Comment:
This isn't a loop anymore (the `break` is unconditional) and can be simplified to a single `if`.

Reply:
Thanks for pointing that out